app/models/proxy.rb

```diff
@@ -528,50 +529,73 @@ def configuration_is_object
 
   class PoliciesConfig
     include ActiveModel::Validations
+    include Enumerable
 
     delegate :each, to: :policies_config
     attr_reader :policies_config
+    protected :policies_config
 
     validate :policies_configs_are_correct
 
     def initialize(policies_config)
-      @policies_config = policies_config.map { |attrs| PolicyConfig.new(attrs) }
+      parsed = policies_config.presence || []
+      parsed = parsed.is_a?(Array) ? parsed.as_json : Array(JSON.parse(parsed))
+
+      @policies_config = policies_normalize(parsed)
+    rescue JSON::ParserError
+      @policies_config = policies_config
     end
 
     def self.name
       'PoliciesConfig'
     end
 
+    def to_json(*args)
+      map(&:to_h).to_json(*args)
+    end
+
+    def find_by(name:, version:)
+      find { |config| config.matches?(name: name, version: version) }
+    end
+    alias find_policy_config_by find_by
+
+    def chain
+      select(&:enabled).map(&:to_h).map { |policy| policy.slice('name', 'version', 'configuration') }
+    end
+    alias policy_chain chain
+
+    def ==(other)
+      self.class == other.class &&
+        policies_config == other.policies_config
+    end
+    alias eql? ==
+
     private
 
+    def policies_normalize(parsed_policies)
+      policies = parsed_policies.map { |attrs| PolicyConfig.new(attrs) }
+      policies.tap { |list| list.push(PolicyConfig.create_default) if policies.none?(&:default?) }
+    end
+
     def policies_configs_are_correct
-      policies_config.each do |policy_config|
-        # TODO: 5: errors.merge!(policy_config.errors)
-        policy_config.errors.each { |attribute, message| errors.add(attribute, message) } unless policy_config.valid?
+      # TODO: 5: errors.merge!(policy_config.errors)
+      reject(&:valid?).map(&:errors).each do |policy_errors|
```

Comment:
`select(&:invalid?)` reads better (just avoiding negative term in the second place)